### ISSUE TITLE
[zscaler_zia] Logs with http_proxy protocol break uri_parts processor

### ISSUE THREAD

**Issue by taylor-swanson** [elastic/integrations#3987]
Logs that have the protocol `http_proxy` will break the `uri_parts` processor in the `web` data stream. As an example, 

- `network.protocol` is `http_proxy` (originally from `json.protocol`)
- `json.eurl` is `www.google.com`

The URI becomes `http_proxy://www.google.com` and this is considered invalid by `uri_parts`. The solution here will be to remap `http_proxy` to the valid protocol `http`.

**Comment by andrewkroh** [comment]
I think the scheme as added into the URI to workaround an earlier problem where `uri_parts` was getting the field extraction wrong when the scheme was missing. Maybe that wasn't the best solution given there are more protocols that we know at that time.

What is the error from the uri_parts processor? I wonder if these other `protocol` values will cause problems.

```
dnsoverhttps
http
http_proxy
https
ssl
tunnel
tunnel_ssl
```

Is there any RFC that states what valid schemes are for URIs? I think it would be reasonable to request that the uri_parts processor accept any arbitrary scheme that's valid under the relevant URI RFC.

**Comment by legoguy1000** [comment]
> Is there any RFC that states what valid schemes are for URIs? I think it would be reasonable to request that the uri_parts processor accept any arbitrary scheme that's valid under the relevant URI RFC.

Yes, https://www.ietf.org/rfc/rfc2396.txt. Section 
```
3.1. Scheme Component

   Just as there are many different methods of access to resources,
   there are a variety of schemes for identifying such resources.  The
   URI syntax consists of a sequence of components separated by reserved
   characters, with the first component defining the semantics for the
   remainder of the URI string.

   Scheme names consist of a sequence of characters beginning with a
   lower case letter and followed by any combination of lower case
   letters, digits, plus ("+"), period ("."), or hyphen ("-").  For
   resiliency, programs interpreting URI should treat upper case letters
   as equivalent to lower case in scheme names (e.g., allow "HTTP" as
   well as "http").

      scheme        = alpha *( alpha | digit | "+" | "-" | "." )

   Relative URI references are distinguished from absolute URI in that
   they do not begin with a scheme name.  Instead, the scheme is
   inherited from the base URI, as described in Section 5.2.
```

**Comment by taylor-swanson** [comment]
It's also worth noting that in the case of tunneled connections (i.e. http_proxy and http method is `CONNECT`), the "url" in the log is just the host:port of the destination server the the proxy will connect to, it's not actually a URL. That comes later, but the proxy won't have any insight into it (it's only known between client and server, and that's especially true when HTTPS is involved). That's probably the case with the other protocols, but I don't have any logs for those.

To summarize, treating this as a URL would be a misnomer in certain cases. I'm between not processing the URL unless the protocol is known to be correct (`http` or `https`), or rewriting the protocol to something valid (`https`) and parsing it so we don't lose that information.
